modules/test_user_dashboard/app/services/test_user_dashboard/create_test_user_account.rb

```diff
@@ -12,9 +12,15 @@ def call
       account = Account.create(idme_uuid: row.delete('idme_uuid'))
       birth_date = Date.parse(row.delete('birth_date')) if row.key?('birth_date')
       services = row.delete('services').split(',')
-      TudAccount.create!(row.merge(account_uuid: account.uuid,
-                                   birth_date: birth_date,
-                                   services: services))
+      account_details = row.merge(account_uuid: account.uuid,
+                                  birth_date: birth_date,
+                                  services: services).compact
+
+      if (tud_account = TudAccount.where(email: row['email']).first)
+        tud_account.update!(account_details)
+      else
+        TudAccount.create!(account_details)
+      end
```

Comment:
This might be a good place for a method like [find_or_initialize_by](https://apidock.com/rails/v4.0.2/ActiveRecord/Relation/find_or_initialize_by)